lib/screens/events/events.dart

```diff
@@ -24,11 +24,12 @@ class _EventsScreenState extends State<EventsScreen> {
           }
         },
         builder: (context, state) {
-          if (state is EventsSuccess) {
+          if (state is EventsInitial) {
+            return _buildInitial();
+          } else if (state is EventsSuccess) {
             json = state.json;
             return _buildSuccess(context);
```

Comment:
you should pass this `state.json` into the `_buildSuccess` function as a parameter instead of doing this.
```suggestion
            return _buildSuccess(context,state.json);
```